src/openrct2/scripting/ScriptEngine.cpp

```diff
@@ -95,81 +97,86 @@ struct ExpressionStringifier final
     }
 
-    void Stringify(const DukValue& val, bool canStartWithNewLine, int32_t nestLevel)
+    void Stringify(const JSValue val, bool canStartWithNewLine, int32_t nestLevel)
     {
         if (nestLevel >= 8)
         {
             _ss << "[...]";
             return;
         }
 
-        switch (val.type())
-        {
-            case DukValue::Type::UNDEFINED:
-                _ss << "undefined";
-                break;
-            case DukValue::Type::NULLREF:
-                _ss << "null";
-                break;
-            case DukValue::Type::BOOLEAN:
-                StringifyBoolean(val);
-                break;
-            case DukValue::Type::NUMBER:
-                StringifyNumber(val);
-                break;
-            case DukValue::Type::STRING:
-                _ss << "'" << val.as_string() << "'";
-                break;
-            case DukValue::Type::OBJECT:
-                if (val.is_function())
-                {
-                    StringifyFunction(val);
-                }
-                else if (val.is_array())
-                {
-                    StringifyArray(val, canStartWithNewLine, nestLevel);
-                }
-                else
-                {
-                    StringifyObject(val, canStartWithNewLine, nestLevel);
-                }
-                break;
-            case DukValue::Type::BUFFER:
-                _ss << "[Buffer]";
-                break;
-            case DukValue::Type::POINTER:
-                _ss << "[Pointer]";
-                break;
-            case DukValue::Type::LIGHTFUNC:
-                _ss << "[LightFunc]";
-                break;
+        if (JS_IsUndefined(val))
+        {
+            _ss << "undefined";
+        }
+        else if (JS_IsNull(val))
+        {
+            _ss << "null";
+        }
+        else if (JS_IsUninitialized(val))
+        {
+            _ss << "uninitialized";
+        }
+        else if (JS_IsBool(val))
+        {
+            _ss << (JS_VALUE_GET_BOOL(val) ? "true" : "false");
+        }
+        else if (JS_IsNumber(val))
+        {
+            StringifyNumber(val);
+        }
+        else if (JS_IsString(val))
+        {
+            StringifyString(val);
+        }
+        else if (JS_IsObject(val))
+        {
+            if (JS_IsFunction(_context, val))
+            {
+                StringifyFunction(val);
+            }
+            else if (JS_IsArray(_context, val))
+            {
+                StringifyArray(val, canStartWithNewLine, nestLevel);
+            }
+            else if (JS_IsError(_context, val))
+            {
+                StringifyError(val);
+            }
+            else
+            {
+                StringifyObject(val, canStartWithNewLine, nestLevel);
+            }
+        }
+        else if (JS_IsArrayBuffer(val))
+        {
+            _ss << "[Buffer]";
+        }
+        else
+        {
+            _ss << "[Unkown Value]";
```

Comment:
Fwiw, typo (`Unkown` instead of `Unknown`)

Reply:
Classic typo. Fixed.